### DISCUSSION TITLE
Redis changes it's license to source available

### DISCUSSION THREAD

**Discussion by hairmare** [argoproj/argo-cd#17599] [General]
I'm not sure how to start but I'm sure that I'm not the first one thinking about this.

There is a high chance that the recent [Redis license change](https://redis.com/blog/redis-adopts-dual-source-available-licensing/) will impact Argo CD, given how we depend on Redis heavily as a caching layer. With the change Redis effectively ceases to be an open source project.

My understanding of our requirements are that, several alternative implementations exist, that would suffice for Argo CD. But that's a very unqualified assessment.

My question is; do we need to raise this issue with the CNCF and start considering alternatives? We are currently using 7.0 from docker hub, and i'm not sure if >=7.4 will we available from that source (given it's a community repo).

Am i wrong with my concerns or do we need to consider this asap?

## Elsewhere

* [Foreman community thread](https://community.theforeman.org/t/redis-adopts-dual-source-available-licensing/37377)

**Comment by hairmare** [comment]
don't feel pressured to answer this quickly btw, We're all aware of KubeCon EU happening in Paris right now; I hope you are all having fun! addressing this change will take a while anyway!

**Comment by hairmare** [comment]
not sure if right process, but: https://github.com/cncf/foundation/issues/750

**Comment by rumstead** [comment]
Although I am not a legal expert and may not fully comprehend the language, my understanding is that the impact is primarily on cloud providers. Some managed Argo CD providers could be affected, but I don't see how it would impact OSS.

> 6. Who is impacted by this change?
> 
> Organizations providing competitive offerings to Redis will no longer be permitted to use new versions of the source code of Redis free of charge under either of the dual licenses. Commercial licensing terms are available and can enable use cases beyond the RSALv2 or SSPLv1 license limitations. If you are building a solution that leverages Redis, but does not specifically compete with Redis itself, there is no impact. If you have specific concerns or questions that you wish to discuss, please email [redis_licensing@redis.com](mailto:redis_licensing@redis.com).


> 17. I am the author of an open-source project that uses Redis in a non-competitive way. If someone else uses my project to produce a competitive product or hosted service (e.g., starts using my project in their SaaS solution), am I at risk of being considered competitive and violating the dual license? Do I need to track all users of my project and report suspected infringing use?
> 
> Only those who are embedding or hosting Redis products in a competitive manner are in violation of the license. The violation does not extend to a project owner who is not doing so and does not require asking others to do so on its behalf. The project owner has no obligation to monitor or report on how others are using their project.

**Reply by hairmare** [reply]
I agree that this change has no impact on most users of Argo CD, right now.

It mainly needs addressing because using Redis is no longer compliant with the [CNCF allow 3rd party license policy](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md). This will also be an issue in enterprises that take a proactive approach to oss license compliance.

The fact the Redis is mostly going to be dropped or replaced in major distros could have an effect on it's availability in general. As far as i understand the change so far it will have an impact on most downstream providers of Argo CD, managed or unmanaged.

**Reply by jannfis** [reply]
It's not only about CNCF guidelines. At least to me personally, what's clear is, that we as an Free and Open Source project, will not ship nor depend on non-FOSS components to be functional. 

So that rules out including or depending on software which is licensed under the SDDL or similar bad licenses, regardless of the restrictions set out by the CNCF. To me, and I think this applies to all or most of the other maintainers too, it's a very personal matter. Argo CD is Free and Open Source Software, and the heck we will encumber with something that's not.

**Reply by rumstead** [reply]
Thanks for sharing your perspective. The intention of my comment was to confirm my understanding of the license, hope it didn’t come off differently. As a large consumer of Argo CD, it’s reassuring to see the passion with FOSS. It’s definitely a shame to see large open sourced projects, that have benefited from the community, change their licensing. Though, a part of me can empathize when cloud providers see more commercial success than the company that invests the most effort maintaining it.

**Comment by hairmare** [comment]
well,,, feels like we might be going with valkey stewarded by the LF:

https://www.linuxfoundation.org/press/linux-foundation-launches-open-source-valkey-community

**Reply by hairmare** [reply]
[Valkey](https://valkey.io/) has forked the container image that is currently used by Argo CD:
* https://github.com/valkey-io/valkey-container
* https://hub.docker.com/r/valkey/valkey

There is a also [a chance](https://github.com/valkey-io/valkey/issues/65#issuecomment-2027437711) that there will be a Bitnami Helm chart for Valkey. This would be beneficial for the downstream Argo CD Helm charts.

**Comment by hairmare** [comment]
I'm following up with an Enhancement proposal here: https://github.com/argoproj/argo-cd/issues/17892
